### PR TITLE
Pension eligibility

### LINKED ISSUE #10: OpenFisca-Israel rules coding
Coding of israel rules into the OpenFisca-Israel instance

needs #1
needs #4

### PR DESCRIPTION
**Pull request by Br3nda**
closes https://github.com/ServiceInnovationLab/Piccolo/issues/10

* Tax and benefit system evolution.
* Details:
  - Added pensions
  - Added genders (binary)
  - Added gender test for pension eligibility

- - - -
- Add a new feature: Pensions

### CODE REVIEW

**Comment by Br3nda** [comment]
the test should pass after we merge #2

File: openfisca_israel/tests/pensions.yaml [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+- name: Man over 62 when immigrated
+  period: 2018-01
+  input_variables:
+    gender: man
+    pension_contributing_years: 22
+    is_resident: true
+    born_in_israel: false
+    age_at_immigration: 63
+  output_variables:
+    eligible_for_pension: true
```
Comment (Br3nda) [inline openfisca_israel/tests/pensions.yaml]: Ooops. This should be false

File: openfisca_israel/variables/pension.py [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+# Import the Entities specifically defined for this tax and benefit system
+from openfisca_israel.entities import *
+
+
+class eligible_for_pension(Variable):
+    value_type = bool
+    entity = Person
+    definition_period = MONTH
+    label = "Eligiblity for pension"
+    reference = "https://github.com/ServiceInnovationLab/Piccolo/issues/4"
```
Comment (dlouise64) [inline openfisca_israel/variables/pension.py]: @Br3nda is this url correct? Shouldn't we be referencing the legislation?

File: openfisca_israel/variables/pension.py [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+            * (years_contributed >= years_required) \
+            * (person('born_in_israel', period) + person('age_at_immigration', period) < max_age_at_immigration)
+
+
+class pension_eligibility_age(Variable):
+    value_type = int
+    entity = Person
+    definition_period = MONTH
+    label = "Age of eligiblity for pension"
+    reference = "https://github.com/ServiceInnovationLab/Piccolo/issues/4"
```
Comment (dlouise64) [inline openfisca_israel/variables/pension.py]: @Br3nda same here

File: openfisca_israel/tests/pensions.yaml (hunk 1 above)
Comment (dlouise64) [reply openfisca_israel/tests/pensions.yaml]: @Br3nda why should this be false? is it because his `age_at_immigration`should be less than 63?

Comment (Br3nda) [reply openfisca_israel/tests/pensions.yaml]: Yes. I have written the test backwards

File: openfisca_israel/variables/pension.py (hunk 2 above)
Comment (Br3nda) [reply openfisca_israel/variables/pension.py]: You're right, i should link to the policy at least

**Comment by Br3nda** [comment]
I've fixed the tests and thresholds for people who immigrate at age over 60/62

File: openfisca_israel/parameters/benefits/pension/max_age_at_immigration.yaml [hunk 4]
```diff
@@ -0,0 +1,17 @@
+description: Max age at time of immigration, to be eligible for persion
+metadata:
+  reference: https://github.com/ServiceInnovationLab/Piccolo/issues/4
```
Comment (dlouise64) [inline openfisca_israel/parameters/benefits/pension/max_age_at_immigration.yaml]: @Br3nda one more here

Comment (dlouise64) [inline openfisca_israel/parameters/benefits/pension/max_age_at_immigration.yaml]: @Br3nda one more here

**Review by dlouise64 — APPROVED** [review]
@Br3nda LGTM!

**Pull request merged**